internal/server/handler.go

```diff
@@ -183,26 +184,30 @@ func (s server) getPost(w http.ResponseWriter, r *http.Request) {
 	owner, uuid := chi.URLParam(r, "owner"), chi.URLParam(r, "uuid")
 
 	if owner == "" || uuid == "" {
-		writeError(w, http.StatusBadRequest, "invalid owner or uuid")
+		api.WriteError(w, http.StatusBadRequest, "invalid owner or uuid")
 		return
 	}
 
 	post, err := s.s.GetPost(r.Context(), storage.PostID{Owner: owner, UUID: uuid})
 	if err != nil {
-		writeInternalError(getLogger(r.Context()).WithError(err), w, "failed to get post")
+		if errors.Is(err, storage.ErrNotFound) {
+			api.WriteError(w, http.StatusNotFound, "post not found")
+			return
+		}
+		api.WriteInternalError(r.Context(), w, fmt.Sprintf("failed to get post: %s", err.Error()))
 		return
 	}
 
 	profile, err := s.s.GetProfiles(r.Context(), post.Owner)
 	if err != nil {
-		writeInternalError(getLogger(r.Context()), w, err.Error())
+		api.WriteInternalError(r.Context(), w, fmt.Sprintf("failed to get profile: %s", err.Error()))
```

Comment:
No need to fix:
WriteInternalErrorf could be created in the api package.

Reply:
I didn't think about internalErrorf , will add it to the go-api package